### PR TITLE
rgw: multisite sync fairness

### PR DESCRIPTION
**Pull request by ivancich**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/rgw/rgw_bid_lock.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+/*
+ * We're currently only using this in rgw and specifically for
+ * multi-site sync. But this could shift to src/common if other parts
+ * of ceph need it.
```
[the remark this answers is not in the document]
Comment (ivancich) [reply src/rgw/rgw_bid_lock.h]: Good idea!

**Comment by cbodley** [comment]
created https://tracker.ceph.com/issues/41230 to track backports

**Comment by ivancich** [comment]
reresh

**Comment by vumrao** [comment]
ubstale.

**Comment by mattbenjamin** [comment]
unstale

**Comment by vumrao** [comment]
unstale

**Comment by trociny** [comment]
@ivancich @cbodley I see this PR was automatically closed due to inactivity. is it because lack of time/interest for this issue, or may there be some complications reveled when testing the implementation? Is there a chance the work will be resumed in near future?

**Comment by mattbenjamin** [comment]
I think this version of the PR has been superceded by https://github.com/ceph/ceph/pull/45958

**Pull request closed without merging**